test/NuGet.Core.Tests/NuGet.Configuration.Test/SettingsTestUtils.cs

```diff
@@ -37,6 +38,34 @@ public static string RemoveWhitespace(string s)
             return Regex.Replace(s, @"\s+", string.Empty);
         }
 
+        public static bool SequenceDeepEquals<T>(IList<T> settings1, IList<T> settings2) where T : SettingBase
+        {
+            if (settings1 == null && settings2 == null)
+            {
+                return true;
+            }
+
+            if (settings1 == null || settings2 == null)
+            {
+                return false;
+            }
+
+            if (settings1.Count() != settings2.Count())
```

Comment:
You can directly call `ICollection<T>.Count` and avoid an unnecessary LINQ extension method call.

    if (settings1.Count != settings2.Count)